lib/collection/src/wal_delta.rs

```diff
@@ -146,6 +154,11 @@ impl RecoverableWal {
         }
         Ok(())
     }
+
+    // TODO(timvisee): provide a better interface for this, pass param into `lock_and_write` instead?
```

Comment:
Left this here because I want to implement this in a better way once I have the time for it. I plan to do so next week.